proposals/XXXX-efficient-token-program.md

```diff
@@ -0,0 +1,97 @@
+```
+Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA invoke [1]
+Program log: Instruction: Transfer
+Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA consumed 249 of 200000 compute units
+Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA success
+```
+Logging the instruction name consumes around `103` compute units, which in some cases reprensents almost the same amount required to execute the instruction. Note that errors will always be logged, since this is considered good use of CU.
+
+## Alternatives Considered
+
+As an alternative to replace the current version of SPL Token, `p-token` could be deployed to a new address and people can be encouraged to transition to that program. This would hinder its adoption and benefits, since people would be very slow to adopt the new program, if they adopt it at all.
+
+## Impact
+
+The main impact is freeing up block CUs, allowing more transactions to be packed in a block; dapp developers benefit since interacting with the Token program will consume significantly less CUs.
+
+Below is a sample of the CU efficient gained by `p-token` compared to the current SPL Token program.
+
+| Instruction                | CU (`p-token`) | CU (`p-token`) + logging | CU (`spl-token`) |
+|----------------------------|----------------|--------------------------|------------------| 
+| `InitializeMint`           | 118            | 221                      | 2967             |
+| `InitializeAccount`        | 170            | 273                      | 4527             |
+| `InitializeMultisig`       | 239            | 348                      | 2973             |
+| `Transfer`                 | 146            | 249                      | 4645             |
+| `Approve`                  | 150            | 268                      | 2904             |
+| `Revoke`                   | 124            | 237                      | 2677             |
+| `SetAuthority`             | 159            | 261                      | 3167             |
+| `MintTo`                   | 144            | 247                      | 4538             |
+| `Burn`                     | 202            | 316                      | 4753             |
+| `CloseAccount`             | 152            | 255                      | 2916             |
+| `FreezeAccount`            | 170            | 287                      | 4265             |
+| `ThawAccount`              | 169            | 283                      | 4267             |
+
+## Security Considerations
+
+`p-token` must be guaranteed to follow the same instructions and accounts layout, as well as have the same behaviour than the current Token implementation.
+
+Any potential risk will be mitigated by extensive fixture testing, formal verification and audits. Since there are potentially huge economic consequences of this change, the feature will be put to a validator vote.
```

Comment:
Since the work's been done, let's add a link to the fuzzing and fixture testing already in place, and also specify the current status of the audit / formal verification. The audit is scheduled, and the formal verification is underway, correct?